flocker/provision/_install.py

```diff
@@ -34,19 +38,25 @@
 
 ARCHIVE_BUCKET = 'clusterhq-archive'
 
+DISTRO_KEY_SUFFIX = get_package_key_suffix(
+    get_installable_version(version)
+)
+
 CLUSTERHQ_REPO = {
     'fedora-20': "https://s3.amazonaws.com/{archive_bucket}/"
-                 "fedora/clusterhq-release$(rpm -E %dist).noarch.rpm".format(
+                 "{key}/clusterhq-release$(rpm -E %dist).noarch.rpm".format(
                      archive_bucket=ARCHIVE_BUCKET,
+                     key='fedora' + DISTRO_KEY_SUFFIX,
                  ),
     'centos-7': "https://s3.amazonaws.com/{archive_bucket}/"
-                "centos/clusterhq-release$(rpm -E %dist).noarch.rpm".format(
+                "{key}/clusterhq-release$(rpm -E %dist).noarch.rpm".format(
                     archive_bucket=ARCHIVE_BUCKET,
+                    key='centos' + DISTRO_KEY_SUFFIX,
                     ),
-    # FLOC-1828 TODO - use ubuntu rather than ubuntu-testing
     'ubuntu-14.04': 'https://{archive_bucket}.s3.amazonaws.com/'
-                'ubuntu-testing/14.04/$(ARCH)'.format(
-                    archive_bucket=ARCHIVE_BUCKET
+                '{key}/14.04/$(ARCH)'.format(
+                    archive_bucket=ARCHIVE_BUCKET,
+                    key='ubuntu' + DISTRO_KEY_SUFFIX,
```

Comment:
This structure is now highly dependent on the global version value. This makes it difficult to test the output of this structure. The time has come to make a function that receives a distribution and a Flocker version, and returns the repo URL.

This will allow you to add tests to `test_install.py` for release and non-release versions creating correct repo URL's.